Rails 5 support

serggl opened the pull request:
- replaced `before_filter` usages with `before_action`
- added a deprecation warning to `GrapeSwaggerRails.options.before_filter` option

serggl commented:
@dblock this one is a followup to #63. Please advise

dblock commented on lib/grape-swagger-rails.rb:
```diff
@@ -3,10 +3,16 @@
 module GrapeSwaggerRails
   class Options < OpenStruct
     def before_filter(&block)
+      ActiveSupport::Deprecation.warn('this option is deprecated and going to be removed soon. ' \
```
Should probably start with a capital `T` to be consistent and have a period after instead?

serggl replied:
what should be the time period?

dblock replied:
I would change "soon" to "version X.Y.Z", whatever the next major one is.

serggl replied:
Not sure what will be the next major one, so changed to 0.3.0. Is it ok?

dblock commented:
This looks good. We need an UPGRADING document like https://github.com/ruby-grape/grape/blob/master/UPGRADING.md, could you add one please?

You don't have to fight rubocop, if something doesn't make sense (like a line too long), just run `rubocop --auto-gen-config`. My workflow is `rubocop -a`, followed by that.

serggl commented:
- added `UPGRADING.md` file
- added a Rails version check in ApplicationController to determine a proper callback to use (Rails 3 does not support `before_action`)

dblock commented:
Merged, thank you.

For versioning pros, what's should we do wrt versioning for this deprecation, should it be 0.3.0 or is deprecating something ok for a minor increment?

dblock commented:
Also @serggl you seem to know what you're doing, want to help out with this project? Maybe cut the next release? If you're up for it tell me your rubygems email and I'll add you to project collaborators.

serggl commented:
@dblock according to semantic versioning (semver.org) this should be a minor version increment, so I guess this should be 0.3.0

Also yes, I think I can help a bit with this project since I use grape/swagger a lot in my day to day work. My rubygems email is sergey.glukhov@gmail.com

dblock commented:
Welcome @serggl. If you can make a release following https://github.com/ruby-grape/grape-swagger-rails/blob/master/RELEASING.md that'd be great!

dblock commented:
Also @aschuster3 has been very active in this project, keep things publicly visible, I opened https://github.com/ruby-grape/grape-swagger-rails/issues/72 for a release.
